pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -40,6 +49,94 @@ func (h *AnnotationBackend) RecordStatesAsync(ctx context.Context, rule *ngmodel
 	go h.recordAnnotationsSync(ctx, panel, annotations, logger)
 }
 
+func (h *AnnotationBackend) QueryStates(ctx context.Context, query models.HistoryQuery) (*data.Frame, error) {
+	logger := h.log.FromContext(ctx)
+	if query.RuleUID == "" {
+		return nil, fmt.Errorf("ruleUID is required to query annotations")
+	}
+
+	if query.Labels != nil {
+		logger.Warn("Annotation state history backend does not support label queries, ignoring that filter")
+	}
+
+	rq := ngmodels.GetAlertRuleByUIDQuery{
+		UID:   query.RuleUID,
+		OrgID: query.OrgID,
+	}
+	err := h.rules.GetAlertRuleByUID(ctx, &rq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to look up the requested rule")
+	}
+	if rq.Result == nil {
+		return nil, fmt.Errorf("no such rule exists")
+	}
+
+	q := annotations.ItemQuery{
+		AlertId: rq.Result.ID,
+		OrgId:   query.OrgID,
+		From:    query.From.Unix(),
+		To:      query.To.Unix(),
+	}
+	items, err := h.annotations.Find(ctx, &q)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query annotations for state history: %w", err)
+	}
+
+	frame := data.NewFrame("states")
+
+	// Annotations only support querying for a single rule's history.
+	// Since we are guaranteed to have a single rule, we can return it as a single series.
+	// Also, annotations don't store labels in a strongly defined format. They are formatted into the label's text.
+	// We are not guaranteed that a given annotation has parseable text, so we instead use the entire text as an opaque value.
+
+	lbls := data.Labels(map[string]string{
+		"from":    "state-history",
+		"ruleUID": fmt.Sprint(query.RuleUID),
+	})
+
+	// TODO: In the future, we probably want to have one series per unique text string, instead. For simplicity, let's just make it a new column.
+	//
+	// TODO: This is a really naive mapping that will evolve in the next couple changes.
+	// TODO: It will converge over time with the other implementations.
```

Comment:
In order for the interface to have a valid implementation, we need **some** way of mapping the annotations into a dataframe.

The approach used here is a very naive one, that really only works for annotations as it only supports a single series generated by a single rule. We have a separate task to create a more permanent mapping, which will be done in the future. The goal here is only get _something_ to show up, in dataframe format.